vic/drivers/shared_all/src/put_data.c

```diff
@@ -580,17 +554,25 @@ put_data(all_vars_struct   *all_vars,
     }
     storage += out_data[OUT_SWE].data[0] + out_data[OUT_SNOW_CANOPY].data[0] +
                out_data[OUT_WDEW].data[0] + out_data[OUT_SURFSTOR].data[0];
-    out_data[OUT_WATER_ERROR].data[0] = calc_water_balance_error(rec, inflow,
-                                                                 outflow,
-                                                                 storage);
+    if (rec > 0) {
+        out_data[OUT_WATER_ERROR].data[0] = calc_water_balance_error(inflow,
+                                                                     outflow,
+                                                                     storage,
+                                                                     save_data->total_moist_storage);
+    }
+    else {
+        out_data[OUT_WATER_ERROR].data[0] = 0.;
```

Comment:
Should this be `MISSING` or `NODATA`? I presume a value cannot be calculated because you do not have a previous storage value?